### PR TITLE
Support generate hub kubeconfigon on the management cluster for addons

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>
If an addon is running in Hosted mode(the addon agent is running on the hosting cluster), we need to generate the hub kubeconfig secret on the hosting cluster.
Hosted mode addon design doc: https://github.com/open-cluster-management-io/enhancements/pull/65

### CODE REVIEW

File: pkg/spoke/addon/configuration.go [hunk 1]
```diff
@@ -60,13 +62,25 @@ func getAddOnInstallationNamespace(addOn *addonv1alpha1.ManagedClusterAddOn) str
 	return installationNamespace
 }
 
+// getAddonInstallMode returns addon installation mode, mode could be `Default` or `Hosted`
+func getAddonInstallMode(addOn *addonv1alpha1.ManagedClusterAddOn) string {
```
Comment (qiujian16) [inline pkg/spoke/addon/configuration.go]: return a boolean is sufficient. You could call it isAddonRunOnManagementCluster etc.

File: pkg/spoke/addon/registration_controller.go [hunk 2]
```diff
@@ -175,7 +178,15 @@ func (c *addOnRegistrationController) syncAddOn(ctx context.Context, syncCtx fac
 // startRegistration starts a client certificate controller with the given config
 func (c *addOnRegistrationController) startRegistration(ctx context.Context, config registrationConfig) context.CancelFunc {
 	ctx, stopFunc := context.WithCancel(ctx)
-	kubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(c.spokeKubeClient, 10*time.Minute, informers.WithNamespace(config.installationNamespace))
+
+	var kubeClient kubernetes.Interface = c.spokeKubeClient
+	if config.installMode == "Hosted" {
```
Comment (qiujian16) [inline pkg/spoke/addon/registration_controller.go]: it needs more doc here. Even a link to the specific design doc.

File: pkg/spoke/addon/registration_controller.go [hunk 3]
```diff
@@ -175,7 +178,15 @@ func (c *addOnRegistrationController) syncAddOn(ctx context.Context, syncCtx fac
 // startRegistration starts a client certificate controller with the given config
 func (c *addOnRegistrationController) startRegistration(ctx context.Context, config registrationConfig) context.CancelFunc {
 	ctx, stopFunc := context.WithCancel(ctx)
-	kubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(c.spokeKubeClient, 10*time.Minute, informers.WithNamespace(config.installationNamespace))
+
+	var kubeClient kubernetes.Interface = c.spokeKubeClient
```
Comment (qiujian16) [inline pkg/spoke/addon/registration_controller.go:187]: I would call it managementKubeClient here, It needs some explanation on why not directly use managedKubeClient.

File: pkg/spoke/addon/configuration.go (hunk 1 above)
Comment (zhujian7) [reply pkg/spoke/addon/configuration.go]: change it to `isAddonRunningOutsideManagedCluster` since it is more like a description of hosted mode

**Pull request merged**